### PR TITLE
Bounty + Question 

### PR DESCRIPTION
**Pull request by yattias**
Allow users to add a bounty during question creation process

https://www.loom.com/share/84e75e7f92a24c1a9f0a2d2945731cc0

### CODE REVIEW

File: components/Bounty/BountyInput.tsx
Comment (yattias) [inline components/Bounty/BountyInput.tsx:1]: This is a refactor I did to extract the "Input part" out of the `BountyModal` component. I have yet to refactor to BountyModal component due to time constraints but that would be the next step here. This is 98% existing code from `BountyModal` component.

File: components/Question/AskQuestionForm.tsx [hunk 1]
```diff
@@ -177,18 +232,65 @@ function AskQuestionForm({ post, user, onExit }: AskQuestionFormProps) {
...
+              <div onClick={() => setWithBounty(false)}>
+                <Button size="small" customButtonStyle={styles.removeBountyBtn}>
+                  <FontAwesomeIcon icon={faPlus} style={{ marginRight: 4 }} />
+                  Remove Bounty
+                </Button>
+              </div>
+            ) : (
+              <div onClick={() => setWithBounty(true)}>
+                <Button size="small" customButtonStyle={styles.addBountyBtn}>
+                  <FontAwesomeIcon icon={faPlus} style={{ marginRight: 4 }} />
```
Comment (manveerxyz) [inline components/Question/AskQuestionForm.tsx:254]: nit: `icon` should be `faTimes` if `withBounty === true`

File: components/Question/AskQuestionForm.tsx [hunk 2]
```diff
@@ -73,6 +77,14 @@ function AskQuestionForm({ post, user, onExit }: AskQuestionFormProps) {
   const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
   const currentUser = useCurrentUser();
   const documentContext = useContext(DocumentContext);
+  const [withBounty, setWithBounty] = useState<boolean>(false);
+  const [bountyOffered, setBountyOffered] = useState<number>(0);
+  const [bountyError, setBountyError] = useState<any>(null);
+  const alert = useAlert();
+  const handleBountyInputChange = ({ hasError, errorMsg, value }) => {
+    setBountyOffered(value);
+    setBountyError(errorMsg);
```
Comment (manveerxyz) [inline components/Question/AskQuestionForm.tsx:86]: it doesn't seem like we show the error message anywhere, e.g. if I don't have enough balance, or am below the minimum required.

Currently if there's an error in the `BountyInput` I just see a disabled button and have to guess why it's not letting me submit.

**Review by manveerxyz — COMMENTED** [review]
Looks good for the most part! Just small comments.

File: components/Question/AskQuestionForm.tsx (hunk 1 above)
Comment (yattias) [reply components/Question/AskQuestionForm.tsx:254]: Good call

File: components/Question/AskQuestionForm.tsx (hunk 2 above)
Comment (yattias) [reply components/Question/AskQuestionForm.tsx:86]: Thanks for letting me know. I did a rebase and I think that might have gotten messed up. There should be an error in there.

**Pull request merged**